contracts/governance/KyberGovernance.sol

```diff
@@ -234,64 +312,84 @@ contract KyberGovernance is Ownable, IKyberGovernance {
+   * @param choice the bit mask of voted options
    * @param v v part of the voter signature
    * @param r r part of the voter signature
    * @param s s part of the voter signature
    **/
   function submitVoteBySignature(
     uint256 proposalId,
-    bool support,
+    uint256 choice,
     uint8 v,
     bytes32 r,
     bytes32 s
   ) external override {
     bytes32 digest = keccak256(
       abi.encodePacked(
         '\x19\x01',
-        keccak256(
-          abi.encode(DOMAIN_TYPEHASH, keccak256(bytes(NAME)), getChainId(), address(this))
-        ),
-        keccak256(abi.encode(VOTE_EMITTED_TYPEHASH, proposalId, support))
+        keccak256(abi.encode(DOMAIN_TYPEHASH, keccak256(bytes(NAME)), getChainId(), address(this))),
+        keccak256(abi.encode(VOTE_EMITTED_TYPEHASH, proposalId, choice))
       )
     );
     address signer = ecrecover(digest, v, r, s);
     require(signer != address(0), 'INVALID_SIGNATURE');
-    return _submitVote(signer, proposalId, support);
-  }
-
-  /**
-   * @dev Set new GovernanceStrategy
-   * Note: owner should be a timelocked executor, so needs to make a proposal
-   * @param governanceStrategy new Address of the GovernanceStrategy contract
-   **/
-  function setGovernanceStrategy(address governanceStrategy) external override onlyOwner {
-    _setGovernanceStrategy(governanceStrategy);
+    return _submitVote(signer, proposalId, choice);
   }
 
   /**
-   * @dev Set new Voting Delay (delay before a newly created proposal can be voted on)
-   * Note: owner should be a timelocked executor, so needs to make a proposal
-   * @param votingDelay new voting delay in terms of blocks
+   * @dev Function to handle voting power changed for a staker
+   *  caller must be the voting power strategy of the proposal
+   * @param staker address that has changed the voting power
+   * @param newVotingPower new voting power of that address,
+   *   old voting power can be taken from records
+   * @param proposalIds list proposal ids that belongs to this voting power strategy
+   *   should update the voteCound of the active proposals in the list
    **/
-  function setVotingDelay(uint256 votingDelay) external override onlyOwner {
-    _setVotingDelay(votingDelay);
+  function handleVotingPowerChanged(
+    address staker,
+    uint256 newVotingPower,
+    uint256[] calldata proposalIds
+  ) external override {
+    for(uint256 i = 0; i < proposalIds.length; i++ ) {
+      if (getProposalState(proposalIds[i]) == ProposalState.Active) {
+        ProposalWithoutVote storage proposal = _proposals[proposalIds[i]];
+        require(address(proposal.strategy) == msg.sender, "ONLY_VOTING_POWER_STRATEGY");
+        Vote memory vote = proposal.votes[staker];
+        if (vote.choice == 0) continue; // not voted yet
+        uint256 oldVotingPower = uint256(vote.votingPower);
+        proposal.totalVotes = proposal.totalVotes + newVotingPower - oldVotingPower;
+        // update vote count for each voted option
```

Comment:
add a comment that proposal is binary then `proposal.options.length` is 2, and choice is 1 or 2

Reply:
it is added in the submitVote, here is handleVotingPowerChanges, so it shouldn't care about that